Compilation error: unexpected ';'

<!-- It is advisable to update all relevant modules using `v outdated` and `v install` -->
**V doctor:**
```
OS: macos, macOS, 11.6, 20G165
Processor: 12 cpus, 64bit, little endian, Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz
CC version: Apple clang version 13.0.0 (clang-1300.0.29.3)

getwd: /Users/tcoynemorgan/github/eev
vmodules: /Users/tcoynemorgan/.vmodules
vroot: /usr/local/Cellar/vlang/0.2.4/libexec
vexe: /usr/local/Cellar/vlang/0.2.4/libexec/v
vexe mtime: 2022-01-31 12:11:00
is vroot writable: true
is vmodules writable: true
V full version: V 0.2.4 46a096b.b34860e

Git version: git version 2.30.1 (Apple Git-130)
Git vroot status: weekly.2022.05
.git/config present: true
thirdparty/tcc status: thirdparty-macos-amd64 93fa86be

```

**What did you do?**
`v -g -o vdbg cmd/v && vdbg eev.v`
```v
import vweb
import sqlite

struct App {
	vweb.Context
pub mut:
	db sqlite.DB
}

fn main() {
	mut app := App{
		// db: sqlite.connect('eev.db') or { panic(err) }
		db: sqlite.connect(':memory:') or { panic(err) }
	}
	sql app.db {
		create table Movie
	}
	vweb.run(app, 8081)
}


struct Movie {
	id          int    [primary; sql: serial]
	name        string
	rating      f32
	tagline     string
	summary     string
	description string
	hash        string
	imdbid      int
	tmdbid      int
	video_id    int
}

['/movie/:movie_id']
fn (mut app App) movie(movie_id int) vweb.Result {
	return app.json(sql app.db {
		select from Movie where id == movie_id
	})
}

```

**What did you expect to see?**

Basic web server with 1 endpoint


**What did you see instead?**
```
==================
   ^~~~~~~~~~~~~~~
/tmp/v_502/eev.12722908764478024433.tmp.c:46405:74: error: unexpected ';' before ')'
                vweb__Result _t1 = vweb__Context_json_T_main__Movie(&app->Context,  _t4;
                                                                                       ^
56 warnings and 1 error generated.
...
==================
(Use `v -cg` to print the entire error message)

builder error: 
==================
C error. This should never happen.

This is a compiler bug, please report it using `v bug file.v`.

https://github.com/vlang/v/issues/new/choose

You can also use #help on Discord: https://discord.gg/vlang

```

@JalonSolov also mentioned on discord ([here](https://discord.com/channels/592103645835821068/592114487759470596/937708851992616970)) that they are experience a similar issue when trying to execute this code on Linux too.

> Weird... on Linux, I get
> `/tmp/v_1000/eev.16565777298037034507.tmp.c:46480: error: ',' expected (got ";")`
> and it is complaining about this line in the output, as it should...
>   `vweb__Result _t1 = vweb__Context_json_T_main__Movie(&app->Context,  _t4;`

--- a/cmd/tools/vtest-self.v
+++ b/cmd/tools/vtest-self.v
@@ -45,7 +45,7 @@ const (
 		'vlib/sqlite/sqlite_orm_test.v',
 		'vlib/v/tests/orm_sub_struct_test.v',
 		'vlib/v/tests/orm_sub_array_struct_test.v',
-		'vlib/v/tests/sql_statement_inside_fn_call_test.v'
+		'vlib/v/tests/sql_statement_inside_fn_call_test.v',
 		'vlib/vweb/tests/vweb_test.v',
 		'vlib/vweb/request_test.v',
 		'vlib/net/http/request_test.v',
@@ -84,7 +84,7 @@ const (
 		'vlib/orm/orm_test.v',
 		'vlib/v/tests/orm_sub_struct_test.v',
 		'vlib/v/tests/orm_sub_array_struct_test.v',
-		'vlib/v/tests/sql_statement_inside_fn_call_test.v'
+		'vlib/v/tests/sql_statement_inside_fn_call_test.v',
 		'vlib/clipboard/clipboard_test.v',
 		'vlib/vweb/tests/vweb_test.v',
 		'vlib/vweb/request_test.v',

--- a/cmd/tools/vtest-self.v
+++ b/cmd/tools/vtest-self.v
@@ -45,6 +45,7 @@ const (
 		'vlib/sqlite/sqlite_orm_test.v',
 		'vlib/v/tests/orm_sub_struct_test.v',
 		'vlib/v/tests/orm_sub_array_struct_test.v',
+		'vlib/v/tests/sql_statement_inside_fn_call_test.v'
 		'vlib/vweb/tests/vweb_test.v',
 		'vlib/vweb/request_test.v',
 		'vlib/net/http/request_test.v',
@@ -83,6 +84,7 @@ const (
 		'vlib/orm/orm_test.v',
 		'vlib/v/tests/orm_sub_struct_test.v',
 		'vlib/v/tests/orm_sub_array_struct_test.v',
+		'vlib/v/tests/sql_statement_inside_fn_call_test.v'
 		'vlib/clipboard/clipboard_test.v',
 		'vlib/vweb/tests/vweb_test.v',
 		'vlib/vweb/request_test.v',

--- a/vlib/v/gen/c/sql.v
+++ b/vlib/v/gen/c/sql.v
@@ -759,7 +759,9 @@ fn (mut g Gen) sql_select(node ast.SqlExpr, expr string, left string) {
 		if node.is_array {
 			g.write('_array')
 		}
-		g.writeln(';')
+		if !g.inside_call {
+			g.writeln(';')
+		}
 	}
 }
 

--- a/vlib/v/tests/sql_statement_inside_fn_call_test.v
+++ b/vlib/v/tests/sql_statement_inside_fn_call_test.v
@@ -1,0 +1,24 @@
+import sqlite
+
+struct Movie {
+	id   int    [primary]
+	name string
+}
+
+fn x(m Movie) int {
+	return m.id
+}
+
+fn test_sql_statement_inside_fn_call() {
+	db := sqlite.connect(':memory:') or { panic('failed') }
+	sql db {
+		create table Movie
+	}
+	m := Movie{1, 'Maria'}
+	sql db {
+		insert m into Movie
+	}
+	dump(x(sql db {
+		select from Movie where id == 1
+	}))
+}

